File: cpp/119.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  vector<int> getRow(int rowIndex) {
    vector<vector<int>> mat(rowIndex + 1);

    for (int i = 0; i < mat.size(); i++) {
      mat[i].resize(i + 1);
      mat[i][0] = mat[i][i] = 1;

      for (int j = 1; j < i; j++)
        mat[i][j] = mat[i - 1][j - 1] + mat[i - 1][j];
    }

    return mat[rowIndex];
  }
};

File: go/119.go
package main

func getRow(rowIndex int) []int {
	mat := make([][]int, 0)

  var size int = rowIndex + 1
	for i := 0; i < size; i++ {
		tmp := make([]int, i+1)
		tmp[0] = 1
		tmp[i] = 1
		for j := 1; j < i; j++ {
			tmp[j] = mat[i-1][j-1] + mat[i-1][j]
		}
    mat = append(mat, tmp)
	}

	return mat[rowIndex]
}
